### PR TITLE
Python 3.3 fixes and AND/OR mutation operator

### PR DESCRIPTION
**Pull request by atodorov**
I am submitting these as one PR because I've set out to make the BoolConstReplacer operator 3.3 compatible and then run into some issues.

From #163 

> Python 3.3 required a change of version for celery, and is still failing. But could be salvageable.

@abingham, @degustaf - in `cosmic_ray/importing.py` we make use of `importlib.machinery.ModuleSpec` which was introduced in 3.4. I don't think we can/want to backport this to 3.3 so maybe drop the entire 3.3 compatibility mode ? What do you say ?

@abingham - do you have any preference if I should drop my 3.3 related commits since they will be kind of pointless if we drop this version entirely ?

### CODE REVIEW

**Comment by degustaf** [comment]
I have no problem leaving this incompatible with 3.3. In fact, that sounds like the right idea.

I just didn't (don't) understand the internals well enough to make that determination.

**Comment by abingham** [comment]
TL;DR - I'm happy to drop 3.3- support.

I figure that CR is far enough from "complete" that I don't want to spend a lot of time making it compatible with Python versions that will be effectively dead when CR is finally fully armed and operational. We have to muck around with fairly obscure elements of the language that seem likely to change, and we could expend a lot our limited time and energy on mixed compatibility.

Based on that and [this discussion](https://github.com/sixty-north/cosmic-ray/issues/101), I'm happy to limit our backwards compatibility arbitrarily. It should work on 3.5, and 3.4 would be very nice. 3.3 and earlier feels unnecessary (@DRMacIver's points about PyPy being a reasonable counterargument).

**Comment by DRMacIver** [comment]
I actually don't think it's a reasonable counterargument any more! The version of pypy that supports 3.2 was super old and didn't really work properly, and there's a new and fairly well funded effort which should upgrade the version of Python supported. Right now the next version of it will be supporting 3.3, but honestly I'd just wait and see where they get to before worrying about pypy (if you're not planning to support Python 2, which is totally reasonable).

**Comment by atodorov** [comment]
I have removed Python 3.3 from the test matrix but otherwise left the other commits in case we want to revisit 3.3 support sometime in the future.

**Comment by atodorov** [comment]
@abingham sorry for making this PR too big but some of the changes go together.

I have added two more mutation operators. One that removes the not keyword and another one which adds it in front of any test condition (if, assert, while and ifexp). Few notes:

* Travis CI fails due to stevedore not being available for Python nightly.
* For some reason the py.test runner fails with 3 survivors while the previous runner did not. All 3 survivors raise an exception and I suspect there is some problem with that. Will look into it so you can merge this without breaking the master branch.

* in `test_mutation_changes_ast` I have commented out the length comparison of original and mutated nodes b/c the not add/remove operators will change that. I'm not sure if we want to keep track of how many new nodes were added or removed. Maybe file an issue if we want to do it.
* The not keyword is added in front of any test conditions. However it is not added before operands of BoolOp, e.g. A or B will not become A or not B. ATM I don't think this is possible to do. I'd love to have some way to mutate first the left side, then the right side of the operation. Maybe that's another RFE/issue.


Please comment on the latest updates and let me know how to proceed.

**Comment by degustaf** [comment]
@atodorov Looking at the [travis-ci build log](https://travis-ci.org/sixty-north/cosmic-ray/jobs/158751677#L353), it apears that the issue is that stevedore isn't compatible with the nightly build yet.  I would mark that as allowed to fail in the build matrix, so that we can monitor it as the official release of 3.6 gets closer.

**Comment by abingham** [comment]
Should I go ahead and merge this in even with the travis failures? (I'm sorry, but I'm too pressed for time right now to come to my own conclusions!)

**Comment by atodorov** [comment]
@abingham please merge it. Only Pythin nightly job fails, the rest are OK.

**Comment by abingham** [comment]
Looks like it needs a merge.

**Comment by atodorov** [comment]
ready to merge

**Comment by abingham** [comment]
Thanks!
